packages/react-tinacms-github/src/toolbar-plugins/createMarkdownDeleteAction.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import {
+  ActionButton,
+  useCMS,
+  Form,
+  Modal,
+  ModalPopup,
+  ModalHeader,
+  ModalBody,
+  ModalActions,
+} from 'tinacms'
+import { Button } from '@tinacms/styles'
+
+const getTitleDefault = (form: Form) => {
+  return form.name
+}
+
+interface options {
+  getTitle?: (form: Form) => string
+  getFilePath?: (form: Form) => string
+}
+
+export const createMarkdownDeleteAction = (
+  { getTitle = getTitleDefault, getFilePath = getTitleDefault }: options = {
+    getTitle: getTitleDefault,
+    getFilePath: getTitleDefault,
+  }
+) => {
+  const DeleteAction = ({ form }: { form: Form }) => {
+    const cms = useCMS()
+    const [active, setActive] = React.useState(false)
+    const open = () => setActive(true)
+    const close = () => setActive(false)
+    const title = getTitle!(form)
+    const filePath = getFilePath!(form)
```

Comment:
I think I would have just exported DeleteAction instead of making it configurable. In the case of `DeleteAction` in `gatsby-tinacms-git` we we just export the component. In that package the filepath and the title are consistent: `filepath=form.id` and `title=form.label`. I went with this because I know that a form of that type is always going to have the fileRelativePath as the id.

To be clear, I'm not saying you _should_ switch it, I'm just explaining the choice I would have made. 

Did you have a use-case in mind for taking this approach?

Reply:
The reason I want this route was for two reasons
1. I wasn't sure if we are guaranteed 100% of the time that `form.name` would always have the path to the file.
2. I wanted to make sure that folks had a way to set how the title was retrieved
I good use case of this would be if I wasn't the title to be "are you sure you want to delete <title of blog post>"
someone could say
```ts
 const deleteAction = createMarkdownDeleteAction({
    getTitle: (form)=>{return form.values.frontmatter.some.nested.title},
  })
```
and for some reason, if the `form.name was` not not the name of the file then 
```ts
 const deleteAction = createMarkdownDeleteAction({
    getFilePath: (form)=>{return form.values.path.to.file},
  })
we might not need this one but I think the first one would be useful for sure.

Reply:
> I wasn't sure if we are guaranteed 100% of the time that form.name would always have the path to the file.

That's fair. I think it's an okay assumption to make for now.

> I wanted to make sure that folks had a way to set how the title was retrieved

`form.label` would be the best bet there. When using the Sidebar, that's what is displayed in the list of forms. It defaults to the ID but it can be set manually.

Reply:
There may be a problem with `form.label`.  One may have a blog page with a form called `Edit blog page`. If we use form.label the prompt will say `do you want to delete Edit blog page` but if we allow the user to define how the label gets set they can point it to a custom name for the file so the prompt would say like `Do you want to delete <Blog title>`

Reply:
Interesting. That's not really how the form label was intended to be used, but I see your point. Let's keep this for now.

Reply:
the form label also appears at the top of the sidebar when editing the form. When only one form is used on a page, using a label of "Edit post" or similar feels more natural than putting the title of the post there.